src/vivarium/framework/lookup/table.py

```diff
@@ -115,6 +116,87 @@ def __repr__(self) -> str:
         return "InterpolatedTable()"
 
 
+class CategoricalTable:
+    """
+    A callable that selects values from a table based on categorical parameters
+    across an index.
+
+    Attributes
+    ----------
+    data
+        The data from which to build the interpolation.
+    population_view
+        View of the population to be used when the table is called with an
+        index.
+    key_columns
+        Column names to be used as categorical parameters in Interpolation
+        to select between interpolation functions.
+    value_columns
+        Names of value columns to be interpolated over. All non parameter- and
+        key- columns in data.
+    validate
+        Whether to validate the data and column names.
+
+    Notes
+    -----
+    These should not be created directly. Use the `lookup` interface on the
+    :class:`builder <vivarium.framework.engine.Builder>` during setup.
+
+    """
+
+    def __init__(
+        self,
+        data: pd.DataFrame,
+        population_view: PopulationView,
+        key_columns: Union[List[str], Tuple[str]],
+        value_columns: Union[List[str], Tuple[str]],
+        validate: bool,
+    ):
+        self.data = data
+        self.population_view = population_view
+        self.key_columns = key_columns
+        self.value_columns = value_columns
+        self.validate = validate
+
+    def __call__(self, index: pd.Index) -> pd.DataFrame:
+        """Get the mapped values for the rows in ``index``.
+
+        Parameters
+        ----------
+        index
+            Index of the population to interpolate for.
+
+        Returns
+        -------
+            A table with the interpolated values for the population requested.
+        """
+        pop = self.population_view.get(index)
+        del pop["tracked"]
+
+        # specify some numeric type for columns, so they won't be objects but
+        # will be updated with whatever column type it actually is
+        result = pd.DataFrame(index=pop.index, columns=self.value_columns, dtype=np.float64)
```

Comment:
Won't this just get changed to object anyway? So if result["category"] is initialized as float64 but then down below you add non-floag64 objects it will just become object?

Reply:
My understanding is that it will remain float64 as long as all data is of a numeric dtype.